src/main/java/org/jabref/gui/entryeditor/FileAnnotationTab.java

```diff
@@ -267,52 +287,91 @@ private JPanel setUpButtons() {
+        sj.add(Localization.lang("Marking") + ": " + markedTxtArea.getText());
 
         new ClipBoardManager().setClipboardContents(sj.toString());
     }
 
     private void reloadAnnotations() {
         isInitialized = false;
         Arrays.stream(this.getComponents()).forEach(this::remove);
+        fileAnnotationCache.remove(parent.getEntry());
         initializeTab(this);
         this.repaint();
     }
 
 
     /**
-     * Fills the highlight and annotation texts and enables/disables the highlight area if there is no highlighted text
+     * Fills the TextAreas of the content and the highlighted or underlined text with the corresponding text and also
+     * changes the label accordingly.
      *
-     * @param annotation either a text annotation or a highlighting from a pdf
+     * @param annotation either a text annotation or a marking from a PDF
      */
-    private void updateContentAndHighlightTextAreas(final FileAnnotation annotation) {
+    private void updateContentAndMarkedTextAreas(final FileAnnotation annotation) {
+        updateMarkingType(annotation);
 
         if (annotation.hasLinkedAnnotation()) {
             // isPresent() of the optional is already checked in annotation.hasLinkedAnnotation()
-            if (!annotation.getLinkedFileAnnotation().getContent().isEmpty()) {
-                contentTxtArea.setText(annotation.getLinkedFileAnnotation().getContent());
+            if (!delocalize(annotation.getLinkedFileAnnotation().getContent()).isEmpty()) {
+                contentTxtArea.setText(delocalize(annotation.getLinkedFileAnnotation().getContent()));
                 contentTxtArea.setEnabled(true);
             } else {
                 contentTxtArea.setText("N/A");
                 contentTxtArea.setEnabled(false);
             }
 
-            if (annotation.getContent().isEmpty()) {
-                highlightTxtArea.setEnabled(false);
-                highlightTxtArea.setText("JabRef: The highlighted area does not contain any legible text!");
+            if (delocalize(annotation.getContent()).isEmpty()) {
+                markedTxtArea.setEnabled(false);
+                markedTxtArea.setText(Localization.lang("The marked area does not contain any legible text!"));
             } else {
-                highlightTxtArea.setEnabled(true);
-                highlightTxtArea.setText(annotation.getContent());
+                markedTxtArea.setEnabled(true);
+                markedTxtArea.setText(delocalize(annotation.getContent()));
             }
-
         } else {
-            contentTxtArea.setText(annotation.getContent());
-            highlightTxtArea.setText("N/A");
-            highlightTxtArea.setEnabled(false);
+            contentTxtArea.setText(delocalize(annotation.getContent()));
+            contentTxtArea.setEnabled(true);
+            markedTxtArea.setText("N/A");
+            markedTxtArea.setEnabled(false);
+        }
+    }
+
+    private String delocalize(String content) {
```

Comment:
Well, the localization must be done where the class is used instead of the model.
As I introduced more annotation types like underline annotations or polygons, I ran out of symbols for the icon in the annotationList, thus, I have prepended the type of the marking to the title of the marking in the list, so that the user is aware what type of annotation it is.
However, in the actual content field this should not be there any more, as it is shown in the label in front of the TextArea.